assets/js/components/settings/SettingsInactiveModules.js

```diff
@@ -34,15 +35,32 @@ import ThumbsUpSVG from '../../../svg/thumbs-up.svg';
 const { useSelect } = Data;
 
 const SettingsInactiveModules = () => {
+	// We store `initialInactiveSlugs` separately to avoid
+	// layout shifts when activating a module as it would otherwise
+	// cause the activated module to be removed upon activation.
+	const [ initialInactiveSlugs, setInitialInactiveSlugs ] = useState();
 	const modules = useSelect( ( select ) => select( CORE_MODULES ).getModules() );
+	useEffect( () => {
+		// Only set initialInactiveSlugs once, as soon as modules are available.
+		if ( ! modules || initialInactiveSlugs !== undefined ) {
```

Comment:
Pretty minor, but if `modules` are `null` or something, wouldn't that be valid but cause this to still be `! modules`? Not a big deal, but wouldn't it be safer to check for `undefined` only here?

Reply:
`modules` here should only ever be an object or `undefined` (which I suppose is the same for `initialInactiveSlugs`, except it uses an array it's just more obvious that it could be `undefined` here because the initial state is empty).

Either should be fine, and I usually go with whatever seems easiest to read.